drivers/dma/dma_stm32.c

```diff
@@ -118,10 +118,14 @@ static void dma_stm32_irq_handler(const struct device *dev, uint32_t id)
 		if (!stream->hal_override) {
 			dma_stm32_clear_ht(dma, id);
 		}
-		stream->dma_callback(dev, stream->user_data, callback_arg, 0);
+		/* set status to 1 for half-complete transfer */
+		stream->dma_callback(dev, stream->user_data, callback_arg, 1);
 	} else if (stm32_dma_is_tc_irq_active(dma, id)) {
 #ifdef CONFIG_DMAMUX_STM32
-		stream->busy = false;
+		/* Circular buffer never stops receiving as long as peripheral is enabled */
+		if (!stream->cyclic && stream->direction == PERIPHERAL_TO_MEMORY) {
```

Comment:
This currently breaks transmitting with spi_ll_stm32. When calling spi_write() consecutively, I get the following error:
```
<err> dma_stm32: dma stream 1 is busy.
<err> dmamux_stm32: cannot configure the dmamux.
```
stream->busy is not cleared in `dma_stm32_irq_handler` because the direction is `MEMORY_TO_PERIPHERAL` and it is also not cleared in `dma_stm32_stop` because `stm32_dma_is_enabled_stream()` returns false (the hardware channel has already stopped).

Please remove `stream->direction == PERIPHERAL_TO_MEMORY`

Reply:
Btw, shouldn't we also check `config->head_block->source_reload_en` here?